### PR TITLE
chore(docs): improve the why a language explanation in faq and readme

### PR DESCRIPTION
**Pull request by ShaiBer**
Fixes: #2181
Screenshot 2023-05-03 at 22 25 14

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: docs/07-faq/005-why-a-language.md [hunk 1]
```diff
@@ -4,18 +4,79 @@ id: why-a-language
...
+
+**Now comes the cool part**: the code that runs inside the inflight function interacts with the `counter` and the `bucket` objects through their *inflight methods* (`counter.inc()` and `bucket.put()`). These methods can only be called from inflight scopes.
+
+## Very cool, but what here cannot be done by a library or compiler extension?
+
+In existing languages, where there is no way to distinguish between multiple execution phases, it is impossible to naturally represent this idea that an object has methods that can only be executed from within a specific execution phase.
+
+In addition, other parts of Wing's *cloud-oriented* cannot be represented naturally without native support from the language. Wwe've seen some worthy efforts in projects like [Pulumi's Function Serialization](https://www.pulumi.com/docs/intro/concepts/function-serialization/) and [Functionless](https://functionless.org/). But although these efforts come close, they either have to make compromises in safety (e.g. it is possible to capture mutable objects) or in the programming model (e.g. type system is too complex).
```
Comment (ekeren) [inline docs/07-faq/005-why-a-language.md:38]:
```suggestion
In addition, other parts of Wing's *cloud-oriented* cannot be represented naturally without native support from the language. We've seen some worthy efforts in projects like [Pulumi's Function Serialization](https://www.pulumi.com/docs/intro/concepts/function-serialization/) and [Functionless](https://functionless.org/). But although these efforts come close, they either have to make compromises in safety (e.g. it is possible to capture mutable objects) or in the programming model (e.g. type system is too complex).
```

File: docs/07-faq/005-why-a-language.md [hunk 2]
```diff
@@ -4,18 +4,79 @@ id: why-a-language
...
+
+We believe that Wing, a language that natively supports this paradigm, will make using it much easier by streamlining common patterns, in a way that is impossible to accomplish with existing ones. Kind of like what C++ did for object orientation. You could write object-oriented code in C, but you had to work hard to make it work without native support from the language.
+
+## So how does writing code in Wing compare to other solutions?
+
+By natively supporting the cloud-oriented programming paradigm, Wing allows developers to write very few lines of code and let the compiler take care of IAM policies and other cloud mechanics.
+
+Below we've written the same simple demo app in Wing and other solutions. This simple app uses a Function to upload a text file to a Bucket, even without the `Counter` seen above that further complicates matters in other solutions:
+
+import CodeComparison from '../src/components/CodeComparison';
```
Comment (ekeren) [inline docs/07-faq/005-why-a-language.md:48]: What is this?

File: docs/07-faq/005-why-a-language.md [hunk 3]
```diff
@@ -4,18 +4,79 @@ id: why-a-language
...
+<CodeComparison />
+<br/>
+
+**The below table contains the main differences that you can see in the code examples:**
+
+| Feature                    | Wing                  | Pulumi                  | Terraform              | AWS CDK                |
+|----------------------------|-----------------------|-------------------------|------------------------|------------------------|
+| Language                   | Wing                  | YAML + your choice      | HCL + your choice      | Your choice            |
+| Line of code               | 7                     | 98                      | 122                    | 72                     |
+| Cloud Mechanics (IAM, N/W) | Generated automatically from intent | Manual              | Manual                 | Manual                 |
```
Comment (ekeren) [inline docs/07-faq/005-why-a-language.md:59]: What is N/W

File: docs/07-faq/005-why-a-language.md [hunk 4]
```diff
@@ -4,18 +4,79 @@ id: why-a-language
...
+| Extensibility                | Custom resources                              | Custom/dynamic providers                | Custom modules                         | Custom Constructs / Resource Providers |
+| Customizing lower levels     | Compiler plugins to customize generated Terraform | Direct access to cloud-provider APIs | No need since working at low level     | Escape hatches to modify underlying CloudFormation |
+| Local simulation             | Built-in functional local simulator with instant hot reloading | No | No | No |
+| Cross-cloud support          | Same code compiles to different clouds        | No (need to write different code for different clouds) | No (need to write different code for different clouds) | No (need to write different code for different clouds) |
+| Provisioning engine          | Same code compiles to Terraform and AWS-CDK (CloudFormation) | Proprietary | Terraform | CloudFormation (CDKTF is used for Terraform, but you cannot compile the same code to different engines) |
```
Comment (ekeren) [inline docs/07-faq/005-why-a-language.md:69]: I don't understand Wing cell:
> Same code compiles to Terraform and AWS-CDK (CloudFormation)
Maybe something like:
>  Terraform / CloudFormation (Extensible solution that supports adding more engines)

File: docs/07-faq/005-why-a-language.md (hunk 2 above)
Comment (ShaiBer) [reply docs/07-faq/005-why-a-language.md:48]: Look at the screenshot in the PR description

File: docs/07-faq/005-why-a-language.md (hunk 3 above)
Comment (ShaiBer) [reply docs/07-faq/005-why-a-language.md:59]: Network

Comment (ShaiBer) [reply docs/07-faq/005-why-a-language.md:59]: Changed to Networking

File: docs/07-faq/005-why-a-language.md [hunk 5]
```diff
@@ -4,18 +4,79 @@ id: why-a-language
...
+|------------------------------|-----------------------------------------------|-----------------------------------------|----------------------------------------|----------------------------------------|
+| Extensibility                | Custom resources                              | Custom/dynamic providers                | Custom modules                         | Custom Constructs / Resource Providers |
+| Customizing lower levels     | Compiler plugins to customize generated Terraform | Direct access to cloud-provider APIs | No need since working at low level     | Escape hatches to modify underlying CloudFormation |
+| Local simulation             | Built-in functional local simulator with instant hot reloading | No | No | No |
+| Cross-cloud support          | Same code compiles to different clouds        | No (need to write different code for different clouds) | No (need to write different code for different clouds) | No (need to write different code for different clouds) |
```
[the remark this answers is not in the document]
Comment (ShaiBer) [reply docs/07-faq/005-why-a-language.md:68]: OK

File: docs/07-faq/005-why-a-language.md (hunk 4 above)
Comment (ShaiBer) [reply docs/07-faq/005-why-a-language.md:69]: Done

**Comment by staycoolcall911** [comment]
@ShaiBer, we've been discussing how this page will look like for a long time... it was worth waiting - it looks great! I love this comparison widget. Absolutely makes the point!

**Pull request merged**